modules/auxiliary/scanner/sap/sap_router_info_request.rb

```diff
@@ -109,7 +109,7 @@ def run_host(ip)
       print_good("#{host_port} - Connected to saprouter")
       print_good("#{host_port} - Sending ROUTER_ADM packet info request")
       sock.put(ni_packet)
-      packet_len = sock.read(4).unpack('H*')[0].to_i 16
+      packet_len = sock.read(4).to_s.unpack('H*')[0].to_i 16
```

Comment:
If this is `nil`, converting it to a String here just delays the problem. In fact, if I'm reading the `while` loop below, it results in an infinite loop.